src/modules/SimpleSlippageRouter.sol

```diff
@@ -0,0 +1,132 @@
+
+    /**
+     * @notice attempted to carry out mint() tx where resultant minted shares is less than user-specified acceptable minted shares amount.
+     * @param minShares specified acceptable min shares amount.
+     * @param maxAssets specified max assets to spend on mint.
+     * @param actualAssetsQuoted actual amount of assets to come from proposed tx, indicating asset amount not enough for specified shares.
+     */
+    error SimpleSlippageAdaptor__MintMinimumSharesSurpassed(
+        uint256 minShares,
+        uint256 maxAssets,
+        uint256 actualAssetsQuoted
+    );
+
+    /**
+     * @notice attempted to carry out redeem() tx where assets returned (the result of redeeming maxShares) < minimumAssets.
+     * @param maxShares specified acceptable max shares amount.
+     * @param minimumAssets specified minimum amount of assets to be returned.
+     * @param actualAssetsQuoted actual amount of assets to come from proposed tx.
+     */
+    error SimpleSlippageAdaptor__RedeemMaxSharesSurpassed(
+        uint256 maxShares,
+        uint256 minimumAssets,
+        uint256 actualAssetsQuoted
+    );
+
+    /**
+     * @notice deposits assets into specified cellar w/ _minimumShares expected and _deadline specified.
+     * @param _cellar specified cellar to deposit assets into.
+     * @param _assets amount of cellar base assets to deposit.
+     * @param _minimumShares amount of shares required at min from tx.
+     * @param _deadline block.timestamp that tx must be carried out by.
+     */
+    function deposit(Cellar _cellar, uint256 _assets, uint256 _minimumShares, uint64 _deadline) public {
+        if (block.timestamp > _deadline) revert SimpleSlippageAdaptor__ExpiredDeadline(_deadline);
+
+        _cellar.asset().safeTransferFrom(msg.sender, address(this), _assets);
+        uint256 shares = _cellar.previewDeposit(_assets);
+        if (shares < _minimumShares) revert SimpleSlippageAdaptor__DepositMinimumSharesUnmet(_minimumShares, shares);
+        _cellar.asset().approve(address(_cellar), _assets);
+        _cellar.deposit(_assets, msg.sender);
+    }
+
+    /**
+     * @notice withdraws assets as long as tx returns more than _assets and is done before _deadline.
+     * @param _cellar specified cellar to withdraw assets from.
+     * @param _assets amount of cellar base assets to withdraw.
+     * @param _maxShares max amount of shares to redeem from tx.
+     * @param _deadline block.timestamp that tx must be carried out by.
+     */
+    function withdraw(Cellar _cellar, uint256 _assets, uint256 _maxShares, uint64 _deadline) public {
+        if (block.timestamp > _deadline) revert SimpleSlippageAdaptor__ExpiredDeadline(_deadline);
+
+        uint256 shares = _cellar.previewWithdraw(_assets);
+        if (shares > _maxShares) revert SimpleSlippageAdaptor__WithdrawMaxSharesSurpassed(_maxShares, shares);
+
+        _cellar.withdraw(_assets, msg.sender, msg.sender); // NOTE: user needs to approve this contract to send shares
```

Comment:
User needs to approve this contract to `spend` shares

Reply:
Reworded comment to say `spend` instead